metsim/metsim.py

```diff
@@ -326,31 +401,37 @@ def setup_output(self, prototype: xr.Dataset=None):
     def _aggregate_state(self):
         """Aggregate data out of the state file and load it into `met_data`"""
         # Precipitation record
-        begin_record = self.params['start'] - pd.Timedelta("90 days")
-        end_record = self.params['start'] - pd.Timedelta("1 days")
-        record_dates = date_range(begin_record, end_record,
+
+        assert self.state.dims['time'] == 90
```

Comment:
Ah, I didn't catch that's what you were doing with the `state_start` and all that. In that case, I like this.